### DISCUSSION TITLE
pyPBD 2.1.2: pd.getVertices() returned NAN

### DISCUSSION THREAD

**Discussion by blower05** [InteractiveComputerGraphics/PositionBasedDynamics#136] [Q&A] [answered]
Hi Bender,

For the example of armadillo dropping into the rigid body torus, another problem was arisen after the just updating of v2.1.2(or v.2.1.1) of pyPBD. However, before the update(2.0.1),  I still  could get the tetmodel softbody armadillo shape and saw it drop onto the torus though the armadillo has many missing faces loaded from the .OBJ when I tried to use Blender 3.1(python 3.9) to edit the script to obtain the animation. 

`pd = model.getParticles()`
`verts = np.array(pd.getVertices())`

returned:
`verts [[nan nan nan]
 [nan nan nan]
 [nan nan nan]
 ...
 [nan nan nan]
 [nan nan nan]
 [nan nan nan]]`

thanks

**Comment by janbender** [comment]
Can you send me your example? I just tested it in armadillo.py and for me it works fine.

**Comment by blower05** [comment]
Hi Prof. Bender,

I sent the file via replying the github forum email  (InteractiveComputerGraphics/PositionBasedDynamics <reply+A3EN4DVLVVYCHDXP5KSN7NOBG2PBHEVBMXHAAQZZEA@reply.github.com>) 

as the format is not allowed to attached to the Git-hub forum. The filename is "pbd_try_armadillo.blend" with the script file "pypbd_blender_armadillo_obj_forLoop.py" inside

vertr [[nan nan nan]
 [nan nan nan]
 [nan nan nan]
 ...
 [nan nan nan]
 [nan nan nan]
 [nan nan nan]]

Thanks

**Comment by janbender** [comment]
Sorry but I did not get any files.

**Reply by blower05** [reply]
I sent again to your university email address with attachment of the blender file and script.

**Comment by blower05** [comment]
Hi Prof.

May I send you via cc your university email?

***@***.******@***.***>


Thanks

Patrick





________________________________

**Comment by janbender** [comment]
Can you send me just a Python file that I can execute?

**Comment by blower05** [comment]
Yes! this is the python file run inside blender.




________________________________

**Comment by janbender** [comment]
I don't have Blender here. So it would be nice just to have a minimal example.

**Reply by blower05** [reply]
the minimal script is just the original given armdillo.py without change.
I also tried this in regular python and np.array(pd.getVertices()) gives normal numbered array and it worked. 

However, since I am bound to edit and use the bmesh and bpy module in blender for outputting the vertices, but the same np.array(pd.getVertices()) without involving bmesh or bpy still giving NAN and I couldn't think what the problem is!
I 've already upgraded numpy as 1.23. I don't what problem arise due to the change from 2.0.1 to 2.1.2 for the pypbd.
Now, I am stuck to use it in Blender.

**Comment by janbender** [comment]
I don't know what goes wrong when you use it in Blender. Is pypbd properly installed in the Blender Python version?

**Reply by blower05** [reply]
Oh! Prof. Bender,
I finally figured it out. The problem occurred at simModel = 5(XPBD) case.
The armadillo will be NAN and won't appear!

Thanks

**Reply by blower05** [reply]
In addition, for simModel = 3 (strain tet, PBD), the armadillo will dissolve into some strange liquid like "materials"!

**Reply by blower05** [reply]
I further tested the other example like beam, it is also the same for simModel =5 that the model will not appear when 5 is chosen.
It looks the update affected the XPBD model which worked OK before!

Thanks

**Comment by janbender** [comment, marked as answer]
Ah, in the last version the simModel numbers changed and I forgot to update the Python examples. Therefore, the stiffness value in the examples was wrong. Now it should work.

**Reply by blower05** [reply]
Oh! got it and thanks! By the way, what kind of stiffness numbers can make a NAN for those returned vertices? was it a case of division by zero?
